### PR TITLE
added docs for wrap and wrap_fn

### PR DESCRIPTION
**Pull request by Dowwie**

### CODE REVIEW

**Review by DoumanAsh — APPROVED** [review]
I think there is also scope middleware to do.

In addition to that you might want to have simple example for `wrap_fn` too

File: src/app.rs [hunk 1]
```diff
@@ -112,7 +112,26 @@ where
         self
     }
 
-    /// Register a middleware.
+    /// Registers heavyweight Application-level middleware, in the form of a
```
Comment (fafhrd91) [inline src/app.rs]: i am not sure about `heavyweight` and `lightweight`. should we maybe use extra info for clarification. something like "in form of reusable components", and "in-place request/response modification".
i expect at some point `wrap` and `wrap_fn` will be merged together, but at the moment rust's type inference is not powerful enough for that.

**Comment by Dowwie** [comment]
I've updated the wording and added docs for scope-level middleware.

**Review by fafhrd91 — APPROVED** [review]

**Comment by Firstyear** [comment]
@Dowwie Thanks for adding these docs, they look really great! One tiny future nitpick is we probably want to avoid emphasis (*word*) style I think? Anyway, thanks for the contribution!

**Pull request merged**